src/client/Microsoft.Identity.Client.Brokers/WamAdapters.cs

```diff
@@ -0,0 +1,236 @@
+                        $"- No Internet connection \n" +
+                        $"Please see https://aka.ms/msal-net-wam for details about Windows Broker integration";
+                    logger.Verbose($"[WamBroker] WAM_provider_error_{errorCode} {errorMessage}");
+                    serviceException = new MsalServiceException($"WAM_provider_error_{errorCode}", errorMessage);
+                    serviceException.IsRetryable = false;
+                    throw serviceException;
+
+                case (int)ResponseStatus.NetworkTemporarilyUnavailable:
+                case (int)ResponseStatus.NoNetwork:
+                case (int)ResponseStatus.ServerTemporarilyUnavailable:
+                    errorCode = authResult.Error.ErrorCode;
+                    internalErrorCode = (authResult.Error.Tag).ToString(CultureInfo.InvariantCulture);
+                    errorMessage =
+                        $"{WamErrorPrefix} \n" +
+                        $" Error Code: {errorCode} \n" +
+                        $" Error Message: {authResult.Error.Status} \n" +
+                        $" WAM Error Message: {authResult.Error.Context} \n" +
+                        $" Internal Error Code: {internalErrorCode} \n" +
+                        $" Is Retryable: true";
+                    logger.Verbose($"[WamBroker] WAM_network_error_{errorCode} {errorMessage}");
+                    serviceException = new MsalServiceException(errorCode.ToString(), errorMessage);
+                    serviceException.IsRetryable = true;
+                    throw serviceException;
+
+                default:
+                    errorCode = authResult.Error.ErrorCode;
+                    internalErrorCode = (authResult.Error.ErrorCode).ToString(CultureInfo.InvariantCulture);
+                    errorMessage = $"Unknown {authResult.Error} (error code {errorCode}) (internal error code {internalErrorCode})";
+                    logger.Verbose($"[WamBroker] {MsalError.UnknownBrokerError} {errorMessage}");
+                    throw new MsalServiceException(MsalError.UnknownBrokerError, errorMessage);
+            }
+        }
+
+        /// <summary>
+        /// Gets the Common Auth Parameters to be passed to Native Interop
+        /// </summary>
+        /// <param name="authenticationRequestParameters"></param>
+        /// <param name="isMsaPassthrough"></param>
+        /// <returns></returns>
+        public static NativeInterop.AuthParameters GetCommonAuthParameters(
+            AuthenticationRequestParameters authenticationRequestParameters, 
+            bool isMsaPassthrough)
+        {
+            var authParams = new NativeInterop.AuthParameters
+                (authenticationRequestParameters.AppConfig.ClientId,
+                authenticationRequestParameters.Authority.AuthorityInfo.CanonicalAuthority);
+
+            //scopes
+            authParams.RequestedScopes = string.Join(" ", authenticationRequestParameters.Scope);
+
+            //redirect URI
```

Comment:
Add a comment that the WAM redirect URi does not need to be configured by the user. This redirect URI is theoretically used by MSAL C++ to fallback to the browser on its own. I say theoretically because in reality MSAL .NET already checks that WAM is available.